aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -342,18 +343,49 @@ Tensor& eye_out_cpu(Tensor& result, int64_t n, int64_t m) {
 
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ full ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Tensor full(IntArrayRef size, Scalar fill_value, const TensorOptions& options) {
-  if (options.layout() == kSparse) {
-    AT_ERROR("full(...) is not implemented for sparse layout");
+namespace {
+
+// Performs dtype inference for full
+TensorOptions infer_full_options(
+  Scalar fill_value,
+  const TensorOptions& options) {
+
+  if (!options.has_dtype()) {
+    if (fill_value.isIntegral(true)) {
+      TORCH_WARN_ONCE(
+        "Deprecation warning: In PyTorch 1.6 torch.full with a bool or ",
+        "integral fill_value will require a dtype or out argument. ",
```

Comment:
probably worth always being consistent and putting the name of the arguments in backticks, i.e. `out`.

Reply:
Yep. Fixed and made vaguer.

Reply:
Fixed.